util/src/main/java/com/google/cloud/hadoop/util/authorization/StorageRequestSummaryFactory.java

```diff
@@ -0,0 +1,106 @@
+package com.google.cloud.hadoop.util.authorization;
+
+import com.google.api.client.http.json.JsonHttpContent;
+import com.google.api.services.storage.Storage;
+import com.google.api.services.storage.StorageRequest;
+import com.google.api.services.storage.model.ComposeRequest;
+import com.google.api.services.storage.model.StorageObject;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Extract information from StorageRequest to authorize access requests. Encapsulate action, bucket
+ * and object name for resource authorization.
+ */
+public class StorageRequestSummaryFactory {
```

Comment:
It seems like it will be more maintainable if we have per-action method in AuthorizationHandler instead of enum where we pass a single method - it will be simpler to maintain in GCS connector and easier to implement. WDYT?

Reply:
I like that. We can encapsulate the strategy interface in GCS connector. Users only need to implement the behavior for the strategies.